### ISSUE TITLE
npm 5.5: nested node_modules folders are ignored by `pack`.

### ISSUE THREAD

**Issue by danShumway** [npm/npm#18942]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [X] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

``npm pack`` does not bundle nested ``node_modules``. I often take advantage of Node's [``require``](https://nodejs.org/api/modules.html#modules_loading_from_node_modules_folders) behavior to avoid relative paths when referencing other files and tests. I've found it to be a big help with keeping code refactorable and readable.

In npm 5.3 and below this works like a charm, but somewhere before 5.5.1, npm started skipping _any_ ``node_modules`` folder, even if it's not associated with a ``package.json`` and not in the root directory.

This also impacts installs; if I try to publish a package with npm 5.5.1, the resulting package in the repo will be broken. If I publish the package with npm 5.3.0 and then install with npm 5.5.1, everything works great again.

Is this an intentional change?

#### How can the CLI team reproduce the problem?

```sh
npm init -y
mkdir -p src/node_modules
echo "console.log('hello');" > src/node_modules/foo.js

# in npm 5.3.0
tar tf $(npm pack)
# package/package.json
# package/src/node_modules/foo.js

#in npm 5.4.0 up to 5.5.1+
tar tf $(npm pack)
# package/package.json
```

### supporting information:

 - `npm -v` prints: v8.4.0
 - `node -v` prints: 5.5.1
 - `npm config get registry` prints: https://registry.npmjs.org/
 -  OS: Linux (Ubuntu)

**Comment by danShumway** [comment]
Could possibly be related to #17708?

**Comment by danShumway** [comment]
It looks like [9b72c4f42cf1](https://github.com/npm/npm-packlist/commit/9b72c4f42cf161e17a780dff6784305522564d12) in ``npm-packlist`` will likely solve this issue. If I get time in the next few days I'll double check.

**Comment by ghost** [comment]
I'm having the same problem, npm pack does not include the /node_modules directory, using npm v5.3.0 and node v8.6.0.  I am also on Ubuntu

**Comment by danShumway** [comment]
I manually updated npm to the most recent version of ``npm-packlist`` and the problem went away, so I suspect everything will be fine whenever the next version of npm is released.

As far as I know ``npm pack`` has never included the root node_modules directory, but 5.3.0 is including nested ones for me.

**Comment by ghost** [comment]
Awesome thanks that worked!

For more details, I was trying to package the cordova node module directory located here:
node root dir/lib/node_modules.  Inside the cordova directory is its own node_modules directory.

Using **Node v8.6.0 & NPM v5.3.0**, running the _npm pack_ command skipped the node_modules directory from within the cordova directory.  I was able to work around this by moving back to **Node v6.11.4 & NPM v3.10.10** then running npm pack.

I've also now tried danShumway's solution to update npm-packlist for NPM 5.3.0.   I did this by running npm install npm-packlist.  After updating, then running npm pack on the cordova node modules directory, it successfully packed everything including the node_modules directory within cordova.

**Comment by danShumway** [comment]
Pinging to make sure the issue doesn't get auto-closed by a bot before I get a chance to verify.

From what I can tell, this will be fixed in anything post ``v5.1.1`` but I'll double check whenever a release happens.

**Comment by danShumway** [comment]
Tested with npm v5.6.0 and ``npm pack`` now includes nested ``node_modules`` folders :)
